content/blog/eng/from-zero-to-infinite-the-final-keyword.md

```diff
@@ -281,44 +283,44 @@ class TryToBreakImmutableObject extends SomeImmutableObject
     }
 }
 
-$four = new TryToBreakImmutableObject('Pippo');
-echo $four->getValue(); //Pippo
+$four = new TryToBreakImmutableObject('Foo');
+echo $four->getValue(); //Foo
 echo $four->change();
 echo $four->getValue();
 ```
 
 ```PHP Fatal error:  Class BreakImmutableObject may not inherit from final class (SomeImmutableObject)```
 
-try it! **[immutable object] (https://3v4l.org/OsOoW)**
+try it! **[immutable object](https://3v4l.org/3vjQ0)**
 
 
 … Here are some links to deepen the topic
 
-- [Final or not final] (https://github.com/thephpleague/period/issues/54) 
-- [Final classes by default] (https://github.com/symfony/symfony/issues/15233)
-- [Final classes: Open for Extension, Closed for Inheritance] (http://verraes.net/2014/05/final-classes-in-php/)
+- [Final or not final](https://github.com/thephpleague/period/issues/54) 
+- [Final classes by default](https://github.com/symfony/symfony/issues/15233)
+- [Final classes: Open for Extension, Closed for Inheritance](http://verraes.net/2014/05/final-classes-in-php/)
 
 Obviously, it isn’t all puppy dogs and rainbows. Technical literature does not perfectly suit to all needs, especially in our field.
 
-What I want to point out is that in some cases **final** can be a good ally and can help us to simplify the design.  
-Using **final** leads programmers to make a round of additional reasoning, to pay more attention to the use of the class. 
-For example, during review, a diff with the removal of **final keyword** could lead to useful comments to find different solutions
+What I want to point out is that in some cases `final` can be a good ally and can help us to simplify the design.  
+Using `final` leads programmers to make a round of additional reasoning, to pay more attention to the use of the class. 
+For example, during review, a diff with the removal of `final` could lead to useful comments to find different solutions.
 
-A downside of the final keyword is that it is considered too coarse an instrument to be effective.
-A developer can simply remove the **final** keyword and then do wrong things!. 
+A downside of the `final` keyword is that it is considered too coarse an instrument to be effective.
+A developer can simply remove the `final` keyword and then do wrong things!. 
 
-I do not agree with this. I've seen misuse of inheritance many times. Therefore, I think that using the **final** is useful both for beginners and senior developers.
+I do not agree with this. I've seen misuse of inheritance many times. Therefore, I think that using the `final` keyword is useful both for beginners and senior developers.
 
-Another problem is that with some moccking library them don't work with **final class**. 
+Another problem is that with some moccking library them don't work with `final` class. 
```

Comment:
- mocking libraries (una sola C)

- Cosa volevi dire con questa frase? Forse me la sono persa la volta scorsa. Vuoi dire che alcune librerie non funzionano con la classe final? Se sì, la frase giusta è così:
Another problem is that some mocking libraries don't work with `final` class.

Reply:
SI hai capito bene ok correggo